lib/widgets/add_todo.dart

```diff
@@ -16,16 +20,29 @@ class AddTodo extends StatefulWidget {
 
 class _AddTodoState extends State<AddTodo> {
   final _todoController = TextEditingController();
+  final _editedTodo = Todo(
+    task: '',
+    date: null,
+  );
 
   void submitTodo() {
-    final enteredTodo = _todoController.text;
-    final date = DateTime.now().toString();
+    // final enteredTodo = _todoController.text;
+    // final date = existingDate != null ? existingDate : DateTime.now().toString();
+    final todo = Todo(
+      task: _todoController.text,
+      date: _editedTodo.date ?? DateTime.now().toString(),
+    );
 
-    if (enteredTodo.isEmpty) {
+    if (todo.task.isEmpty) {
       return;
     }
+    // String newDate;
+    // String newtodo;
+    _editedTodo.date != null
```

Comment:
I see no setting of a new value for `_editedTodo.date` in this file, so since it is null by default while checking here if it is not null, it will not run the expected method.